pkg/syncer/cnsoperator/controller/cnsregistervolume/util.go

```diff
@@ -161,10 +164,43 @@ func getK8sStorageClassNameWithImmediateBindingModeForPolicy(ctx context.Context
 			"Either storagepolicyId: %s doesn't match any storage class, or the policy is not assigned to namespace: %s",
 			storagePolicyID, namespace)
 	} else {
-		// TODO: Implement Quota check using new Storage Quota CR before returning storage class
-		return scName, nil
+		storagePolicyQuotaList := &cnsstoragepolicyquotasv1alpha1.StoragePolicyQuotaList{}
+		err := client.List(ctx, storagePolicyQuotaList, &ctrlruntimeclient.ListOptions{
+			Namespace: namespace,
+		})
+		if err != nil {
+			return "", logger.LogNewErrorf(log, "Failed to list StoragePolicyQuota CR on the namespace: %s", namespace)
+		}
+		log.Debugf("Found scName %s which has matching storagePolicyId %s", scName, storagePolicyID)
+		log.Debugf("Fetch storagePolicyQuotaList: %+v  in namespace %s", storagePolicyQuotaList, namespace)
```

Comment:
can we remove these log lines if not needed ?

Reply:
This is DEBUG level log and will help us to debug if bugs related to this code. So let us keep it.